src/mono/mono/metadata/native-library.c

```diff
@@ -1323,8 +1323,8 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 
 	/* If qcalls are disabled, we fall back to the normal pinvoke code for them */
 #ifndef DISABLE_QCALLS
-	if (strcmp (new_scope, "QCall") == 0) {
-		piinfo->addr = mono_lookup_pinvoke_qcall_internal (method, status_out);
+	if (strcmp (new_scope, "QCall") == 0 || strcmp(new_scope, "libSystem.Globalization.Native") == 0) {
```

Comment:
why is globalization hardcoded here?  there shouldn't be any special casing.

Reply:
Lookup via `GlobalizationResolveDllImport` should be conditional on `strcmp(new_scope, "libSystem.Globalization.Native")`, otherwise there is a chance of matching a method from a wrong lib if it happens to have the same name as one in Globalization. It is highly unlikely though.

Folding the `GlobalizationResolveDllImport` lookup into handling of `QCall` does not seem right though, especially since there are further plans for QCall code. 

I think I will just add the following:

```c
#if defined(ENABLE_NETCORE) && !defined(NO_GLOBALIZATION_SHIM)
	if (strcmp(new_scope, "libSystem.Globalization.Native") == 0) {
		const void* method_impl = GlobalizationResolveDllImport(new_import);
		if (method_impl)
		{
			return (gpointer)method_impl;
		}
	}
#endif
```

Reply:
Looks like changes suggested above worked. I will consider this resolved. Please let me know if there are concerns.